### PR TITLE
add data integrity to single node upgrade test

### PR DESCRIPTION
**Pull request by moogacs**

### CODE REVIEW

File: apps/upgrade-single-node/create.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import weaviate
+import weaviate.classes as wvc
+
+sphere_file = "sphere.10k.jsonl"
+
+sizes = [10, 100, 1000, 10000, 10, 10, 10, 10, 50]
+
+
+def _import():
+    client = weaviate.connect_to_local()
```
Comment (dirkkul) [inline apps/upgrade-single-node/create.py:13]: with weaviate.connect_to_local() as client:
   ...rest of code....
   
then the client will be closed automatically at the end of the contextmanager

File: apps/upgrade-single-node/query.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import weaviate
+import weaviate.classes as wvc
+
+sphere_file = "sphere.10k.jsonl"
+
+sizes = [10, 100, 1000, 10000, 10, 10, 10, 10, 50]
+
+def query():
+    client = weaviate.connect_to_local()
```
Comment (dirkkul) [inline apps/upgrade-single-node/query.py:12]: same

File: apps/upgrade-single-node/create.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                    if i >= sizes[j]:
+                        break
+
+                    json_parsed = json.loads(jsonl)
+                    batch.add_object(
+                        properties={
+                            "url": json_parsed["url"],
+                            "title": json_parsed["title"],
+                            "raw": json_parsed["raw"],
+                            "sha": json_parsed["sha"],
```
Comment (dirkkul) [inline apps/upgrade-single-node/create.py:47]: can we do this with some dummy data to not have to save the sphere dataset? Shouldn't play a role

File: apps/upgrade-single-node/requirements.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+weaviate-client>=3.11.0
```
Comment (dirkkul) [inline apps/upgrade-single-node/requirements.txt:1]: that would be very old 😅  - use 4.7.1?

File: apps/upgrade-single-node/query.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+                    continue
+                if  i >num_queries*10 or i >= tenant_size:
+                    break
+                json_parsed = json.loads(jsonl)
+                start= time.time()
+                collection.with_tenant(f"Tenant{j}").query.near_vector(json_parsed["vector"])
+                if i == 0:
+                    print((time.time()-start)*1000, "first", tenant_size)
+                if i == num_queries*10 or i == tenant_size-10:
+                    print((time.time() - start) * 1000, "last", tenant_size)
```
Comment (dirkkul) [inline apps/upgrade-single-node/query.py:32]: the prints are not needed here, those were added for query time debugging

File: apps/upgrade-single-node/query.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+sphere_file = "sphere.10k.jsonl"
+
+sizes = [10, 100, 1000, 10000, 10, 10, 10, 10, 50]
+
+def query():
+    client = weaviate.connect_to_local()
+    collection = client.collections.get("MultiTenantPerformanceTest")
+    times: dict[str, float] = {}
+    num_queries= 2000
```
Comment (dirkkul) [inline apps/upgrade-single-node/query.py:15]: let's just do 10

**Review by dirkkul — COMMENTED** [review]
I think you can clean the script up abit more :) There is a lot of debugging vars/prints etc that are not really needed for this test

File: apps/upgrade-single-node/create.py (hunk 3 above)
Comment (moogacs) [reply apps/upgrade-single-node/create.py:46]: i will think about it, but the idea is large data to create a snapshot

**Review by reyreaud-l — APPROVED** [review]

**Pull request merged**